src/Examine.Lucene/Search/LuceneSearchResults.cs

```diff
@@ -1,23 +1,26 @@
-﻿using System;
+using System;
 using System.Collections;
 using System.Collections.Generic;
 
 namespace Examine.Lucene.Search
 {
-    public class LuceneSearchResults : ISearchResults
+    public class LuceneSearchResults : ISearchResults, IFacetResults
     {
-        public static LuceneSearchResults Empty { get; } = new LuceneSearchResults(Array.Empty<ISearchResult>(), 0);
+        public static LuceneSearchResults Empty { get; } = new LuceneSearchResults(Array.Empty<ISearchResult>(), 0, new Dictionary<string, IFacetResult>());
 
         private readonly IReadOnlyCollection<ISearchResult> _results;
 
-        public LuceneSearchResults(IReadOnlyCollection<ISearchResult> results, int totalItemCount)
+        public LuceneSearchResults(IReadOnlyCollection<ISearchResult> results, int totalItemCount, IDictionary<string, IFacetResult> facets)
         {
             _results = results;
             TotalItemCount = totalItemCount;
+            Facets = facets;
         }
 
         public long TotalItemCount { get; }
 
+        public IDictionary<string, IFacetResult> Facets { get; }
```

Comment:
should be IReadOnlyDictionary

Reply:
You might also need to be careful about dictionaries with Lucene and field names not always being the correct case. In some cases within Examine when working with Dictionaries, they must always be constructed with an invariant key option which will probably be required for any dictionaries created for these facet results.

Reply:
I've added:
```csharp
var facets = new Dictionary<string, IFacetResult>(StringComparer.InvariantCultureIgnoreCase);
```
in `LuceneSearchExecutor.ExtractFacets` where the dictionary is created. I'm not sure this was what you meant?

Reply:
Yep that's what I meant :)